sycl/include/CL/sycl/INTEL/esimd/detail/emu/esimdcpu_device_interface.hpp

```diff
@@ -0,0 +1,68 @@
+//==----- esimdcpu_device_interface.hpp - DPC++ Explicit SIMD API ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+/// \file esimdcpu_device_interface.hpp
+/// Declarations for ESIMD_CPU-device specific definitions.
+/// ESIMD intrinsic and LibCM functionalities required by intrinsic defined
+///
+/// This interface is for ESIMD intrinsic emulation implementations
+/// such as slm_access to access ESIMD_CPU specific-support therefore
+/// it has to be defined and shared as include directory
+///
+/// \ingroup sycl_pi_esimd_cpu
+
+#pragma once
+
+#include <CL/sycl/detail/plugin.hpp>
+#include <cstdint>
+
+#define ESIMD_CPU_DEVICE_REQUIRED_VER 0
+
+#ifdef _MSC_VER
+// Definitions for type consistency between ESIMD_CPU and CM_EMU
+typedef unsigned int uint;
+typedef unsigned short ushort;
+typedef unsigned char uchar;
+#endif // _MSC_VER
+
+struct ESIMDDeviceInterface {
+  void *reserved;
+
+  ESIMDDeviceInterface();
+#include "esimd_emu_functions_v0.h"
+};
+
+struct OpaqueDataAccess {
+  uintptr_t version;
+  struct ESIMDDeviceInterface *interface;
+};
+
```

Comment:
plugin opaque data structure definition should go into the plugin interface, as it is supposed to be used by all plugins

Reply:
@kbobrovs , should we define what opaque data looks like including version info in a SYCL-widely shared header? Why don't we let clients/users decide opaque data type? For example, some may prefer string-type version info (like `PiVersion`) over integer-type per their versioning style.

Reply:
> @kbobrovs , should we define what opaque data looks like including version info in a SYCL-widely shared header? Why don't we let clients/users decide opaque data type? For example, some may prefer string-type version info (like `PiVersion`) over integer-type per their versioning style.

As I wrote below, we decided to maintain opaque versioning inside plugins, hence yes, some changes are still needed - I added them as suggestions to this PR.